packages/application-components/src/components/dialogs/internals/dialog-header.tsx

```diff
@@ -21,41 +25,76 @@ export const TextTitle = (props: TTextTitleProps) => (
   </Text.Headline>
 );
 
-type TitleProps = Pick<Props, 'title'>;
+type TitleProps = Pick<Props, 'title' | 'aria-label'>;
 const Title = (props: TitleProps) => {
   if (typeof props.title === 'string') {
-    return <TextTitle title={props.title} />;
-  } else {
-    return <>{props.title}</>;
+    return (
+      /* 
+      FIXME: Temporary workaround for https://github.com/radix-ui/primitives/issues/2986#issuecomment-2455201034
+      Radix UI's DialogContent requires a DialogTitle, but we cannot pass TextTitle and use the asChild prop.
+      To meet this requirement and avoid rendering two heading elements with the title in the DOM, we are hiding the DialogTitle.
+      The intended JSX structure should be:
+      ```
+      <DialogTitle asChild>
+        <TextTitle title={props.title} />
+      </DialogTitle>
+      ```
+       */
+      <>
+        <TextTitle title={props.title} />
+        <div
+          css={css`
+            display: none;
+          `}
+        >
+          <DialogTitle>{props.title}</DialogTitle>
+        </div>
+      </>
+    );
   }
+  return (
+    <>
+      {props.title}
+      <VisuallyHidden>
+        <DialogTitle>{props['aria-label']}</DialogTitle>
+      </VisuallyHidden>
+    </>
+  );
 };
 
-const DialogHeader = (props: Props) => (
-  <div
-    css={css`
-      flex: 0 1 auto;
-      display: flex;
-      flex-direction: column;
-      margin-bottom: ${uiKitDesignTokens.spacing30};
-    `}
-  >
-    <Spacings.Inline
-      scale="m"
-      alignItems="center"
-      justifyContent="space-between"
+const DialogHeader = (props: Props) => {
+  useWarning(
+    typeof props.title === 'string' ||
+      (typeof props.title !== 'string' && Boolean(props['aria-label'])),
+    'app-kit/DialogHeader: "aria-label" prop is required when the "title" prop is not a string.'
+  );
```

Comment:
This was moved in here from https://github.com/commercetools/merchant-center-application-kit/blob/19489dad442b365f157771278e41c5c5fe60e19f/packages/application-components/src/components/dialogs/internals/dialog-container.tsx#L65-L69 Before the Modal component accepted a `contentLabel` prop. Now, we need to manually create an accessible title.